src/coreclr/nativeaot/System.Private.TypeLoader/src/Internal/Runtime/TypeLoader/GenericDictionaryCell.cs

```diff
@@ -770,17 +770,17 @@ private bool NeedsDictionaryParameterToCallCanonicalVersion(MethodDesc method)
                 if (Method is NoMetadataMethodDesc)
```

Comment:
We just call `IsStaticMethodSignature` everywhere else in this file, without special-casing `NoMetadataMethodDesc`. Is there a reason why we need to special-case it here?

Reply:
I don't know and I'm worried we don't have enough test coverage to go and touch too much of type loader. I'm already worried about the change I'm making now :(